packages/app-admin-ui/index.js

```diff
@@ -289,45 +252,24 @@ class AdminUIApp {
       })
     );
 
-    const secureMiddleware = webpackDevMiddleware(secureCompiler, webpackMiddlewareConfig);
-    const secureHotMiddleware = webpackHotMiddleware(secureCompiler, webpackHotMiddlewareConfig);
-
+    const middlewares = [
+      { secure: webpackDevMiddleware(secureCompiler, webpackMiddlewareConfig) },
+      { secure: webpackHotMiddleware(secureCompiler, webpackHotMiddlewareConfig) },
+    ];
     if (this.authStrategy) {
+      // override lists so that schema and field views are excluded
       const publicCompiler = webpack(
         getWebpackConfig({
-          // override lists so that schema and field views are excluded
           adminMeta: { ...adminMeta, lists: {} },
           adminViews: this.getAdminViews({ keystone, includeLists: false }),
           entry: 'public',
         })
       );
-
-      const publicMiddleware = webpackDevMiddleware(publicCompiler, webpackMiddlewareConfig);
-      const publicHotMiddleware = webpackHotMiddleware(publicCompiler, webpackHotMiddlewareConfig);
-
-      return [
-        {
-          secure: secureMiddleware,
-          public: publicMiddleware,
-        },
-        {
-          secure: secureHotMiddleware,
-          public: publicHotMiddleware,
-        },
-      ];
-    } else {
-      return [
-        {
-          secure: secureMiddleware,
-        },
-        {
-          secure: secureHotMiddleware,
-        },
-      ];
+      middlewares[0].public = webpackDevMiddleware(publicCompiler, webpackMiddlewareConfig);
+      middlewares[1].public = webpackHotMiddleware(publicCompiler, webpackHotMiddlewareConfig);
```

Comment:
I see that you are doing here, but `middlewares[0].public` etc... kinda sucks. Part of me was thinking if you're going to the trouble to re-write a lot of this why stick with this less readable convention...

But I'm guessing you didn't want to change the shape this exports which would have had other consequences?

Reply:
Yeah, I want this to be functionally identical so I can better understand what's going on before making any functional changes.